apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js

```diff
@@ -4,119 +4,178 @@
-		setState( { isEditing: false } );
-		setAttributes( { templateId: id } );
-	};
+		if ( ! isTemplate && templateId && ! template ) {
+			return (
+				<Placeholder>
+					<Spinner />
+				</Placeholder>
+			);
+		}
 
-	const showToggleButton = ! isEditing || !! templateId;
-	const showPlaceholder = isEditing || ! templateId;
-	const showContent = ! isEditing && !! templateId;
-	const isTemplate = 'wp_template' === fullSiteEditing.editorPostType;
+		useEffect( () => {
+			receiveTemplateBlocks();
+		} );
 
-	const editTemplatePartUrl = addQueryArgs( fullSiteEditing.editTemplatePartBaseUrl, {
-		post: templateId,
-		fse_parent_post: currentPostId,
-	} );
+		const toggleEditing = () => setState( { isEditing: ! isEditing } );
 
-	return (
-		<Fragment>
-			{ showToggleButton && isTemplate && (
-				<BlockControls>
-					<Toolbar>
-						<IconButton
-							className={ classNames( 'components-icon-button components-toolbar__control', {
-								'is-active': isEditing,
-							} ) }
-							label={ __( 'Change Template Part' ) }
-							onClick={ toggleEditing }
-							icon="edit"
-						/>
-					</Toolbar>
-				</BlockControls>
-			) }
-			<div
-				className={ classNames( 'template-block', {
-					[ `align${ align }` ]: align,
-				} ) }
-			>
-				{ showPlaceholder && (
-					<Placeholder
-						icon="layout"
-						label={ __( 'Template Part' ) }
-						instructions={ __( 'Select a template part to display' ) }
-					>
-						<div className="template-block__selector">
-							<PostAutocomplete
-								initialValue={ get( template, [ 'title', 'rendered' ] ) }
-								onSelectPost={ onSelectTemplate }
-								postType="wp_template_part"
+		const onSelectTemplate = ( { id } ) => {
+			setState( { isEditing: false } );
+			setAttributes( { templateId: id } );
+		};
+
+		return (
+			<Fragment>
+				{ showToggleButton && isTemplate && (
+					<BlockControls>
+						<Toolbar>
+							<IconButton
+								className={ classNames( 'components-icon-button components-toolbar__control', {
+									'is-active': isEditing,
+								} ) }
+								label={ __( 'Change Template Part' ) }
+								onClick={ toggleEditing }
+								icon="edit"
 							/>
-							{ !! template && (
-								<a href={ editTemplatePartUrl }>
-									{ sprintf( __( 'Edit "%s"' ), get( template, [ 'title', 'rendered' ], '' ) ) }
-								</a>
-							) }
-						</div>
-					</Placeholder>
+						</Toolbar>
+					</BlockControls>
 				) }
-				{ showContent && (
-					<Fragment>
-						<RawHTML className="template-block__content">
-							{ get( template, [ 'content', 'rendered' ] ) }
-						</RawHTML>
-						{ ! isTemplate && (
-							<Placeholder
-								className="template-block__overlay"
-								instructions={ __(
-									'This block is part of your site template and may appear on multiple pages.'
+				<div
+					className={ classNames( 'template-block', {
+						[ `align${ align }` ]: align,
+					} ) }
+				>
+					{ showPlaceholder && (
```

Comment:
I think this whole block can be removed now - this is a remnant of an earlier phase when we were selecting Template Parts to preview manually. These are now populated automatically based on assigned Page Template.

Reply:
I wasn't really sure we wanted to go down that road, so I left everything there. But I'll take this as approval so I'll remove it :)

Reply:
I think it's unreachable now, unless something in our code is broken. 😄

Reply:
Lots of code removed in 7078ae74866f87e53669ed81b9bb1b853190fa5b.